src/nix/profile.cc

```diff
@@ -390,7 +395,25 @@ struct CmdProfileInstall : InstallablesCommand, MixDefaultProfile
 
             element.updateStorePaths(getEvalStore(), store, res);
 
-            manifest.addElement(std::move(element));
+            auto elementName = getNameFromElement(element);
+
+            // Check if the element already exists.
+            auto existingPair = manifest.elements.find(elementName);
+            if (existingPair != manifest.elements.end()) {
+                auto existingElement = existingPair->second;
+                auto existingSource = existingElement.source;
+                auto elementSource = element.source;
+                if (existingSource
+                    && elementSource
+                    && existingElement.priority == element.priority
+                    && existingSource->originalRef == elementSource->originalRef
+                    && existingSource->attrPath == elementSource->attrPath
+                    ) {
+                    throw Error("'%s' is already installed", elementName);
```

Comment:
Maybe repeated installation should be idempotent, i.e.
```suggestion
                    warn("'%s' is already installed", elementName);
                    continue;
```

This way, `nix profile install X` is defined in terms of the post-condition being that X is installed. That could be convenient for e.g. installation scripts.

Reply:
Ah, yes I can see being less strict would help :+1: I've applied your suggestion. Thanks!